lib/bitcoin/lock/LockResolver.ts

```diff
@@ -85,9 +85,11 @@ export default class LockResolver {
     const lockStartBlock = await this.calculateLockStartingBlock(lockTransaction);
 
     // (C). verify that the lock duration is valid
-    if (!this.isLockDurationValid(lockStartBlock, scriptVerifyResult.unlockAtBlock!)) {
+    const unlockAtBlock = lockStartBlock + scriptVerifyResult.lockDurationInBlocks!;
+
+    if (!this.isLockDurationValid(lockStartBlock, unlockAtBlock)) {

```

Comment:
Since we operate on relative lock now, would i make sense to make this function just take in the `scriptVerifyResult.lockDurationInBlocks` ?